src/cudamatrix/cu-kernels.cu

```diff
@@ -1010,11 +1010,14 @@ static void _trace_mat_mat(const Real* A, const Real* B, MatrixDim dA,
     __syncthreads();
   }
 
-  // Warp reduce. Implicitly synchronized within a warp.
+  // Warp reduce
```

Comment:
Thanks.
I think this reduction is just summing the array.  Surely there must be a cub approach for this-- wouldn't that be a more standard approach?  
I don't know much about this stuff, just want to know.

Reply:
cub may be better if it can be used here.  I will take a look soon.